metricsbp/baseplate_hooks.go

```diff
@@ -11,53 +11,57 @@ const (
 	fail    = "fail"
 )
 
-// BaseplateHook registers each Server Span with a MetricsSpanHook.
-type BaseplateHook struct {
+// CreateServerSpanHook registers each Server Span with a MetricsSpanHook.
+type CreateServerSpanHook struct {
 	Metrics Statsd
 }
 
-// OnServerSpanCreate registers MetricSpanHooks on a Server Span.
-func (h BaseplateHook) OnServerSpanCreate(span *tracing.Span) error {
+// OnCreateServerSpan registers MetricSpanHooks on a server Span.
+func (h CreateServerSpanHook) OnCreateServerSpan(span *tracing.Span) error {
 	serverSpanHook := newSpanHook(h.Metrics, span)
-	span.RegisterHook(serverSpanHook)
+	span.AddHooks(serverSpanHook)
 	return nil
 }
 
 // SpanHook wraps a Span in a Timer and records a "success" or "fail"
 // metric when the Span ends based on whether an error was passed to `span.End`
 // or not.
 type SpanHook struct {
```

Comment:
Do we want to export this type at all? I think the intended use case is that user register the `CreateServerSpanHook`, which will auto register this hook whenever a new server span is created, so is there a use case that user want to register a `SpanHook` manually? If there's a use case for that, then we likely will also need to export `newSpanHook` function, as callers outside of this package has no way to setup the timer (if the caller just construct `SpanHook` without calling `newSpanHook`, `timer` will be nil, and then when `OnPostStart` is called it will panic).

Reply:
Nope, I don't think so, still getting used to the idea of exporting vs not exporting 😄